### ISSUE TITLE
False positive for invalid-name in constant variables modified

### ISSUE THREAD

**Issue by moylop260** [pylint-dev/pylint#10456]
### Bug description

```python
GOOD_CONST_NAME = ""
GOOD_CONST_NAME += " "
```

### Configuration

```ini

```

### Command used

```shell
pylint -d all -e invalid-name my_file.py
```

### Pylint output

```python
Variable name "GOOD_CONST_NAME" doesn't conform to '[a-z_][a-z0-9_]{2,30}$' pattern (invalid-name)
```

### Expected behavior

No raise invalid-name for a good constant name

### Pylint version

```shell
pylint 4.0.0-dev0
astroid 4.0.0a0
Python 3.12.11 (main, Jun  3 2025, 15:41:47) [Clang 17.0.0 (clang-1700.0.13.3)]

Commit 85c8f50f7803e237619b506811cb467b9e77ea93
```

### OS / Environment

MacOSX 15.5 (24F74)

### Additional dependencies

```python

```

**Comment by Pierre-Sassoulas** [comment]
There's been a discussion about this with Jacob somewhere and I can"t find where in a reasonable time, it's a design choice we have to make, I would say pylint is working as intended here, if you modify a variable it's not a constant anymore (but not everyone agree).

**Comment by moylop260** [comment]
It makes sense to me

Thank you for the quick reply
